src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,133 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev,T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            newNode.prev = newNode;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> searchIndex = findNode(index);
+            Node<T> newNode = new Node<>(searchIndex.prev, value, searchIndex);
+            searchIndex.prev.next = newNode;
+            searchIndex.prev = newNode;
+            size++;
+        }
```

Comment:
```suggestion
            Node<T> next = findNode(index);
            Node<T> prev = next.prev;
            Node<T> newNode = new Node<>(prev, value, next);
            if (prev == null) {
               head = newNode;
            } else {
               prev.next = newNode;
            }
            next.prev = newNode;
            size++;
```
you can simplify  this code and avoid code duplication